src/Compilers/Core/Portable/DiagnosticAnalyzer/SymbolDeclaredCompilationEvent.cs

```diff
@@ -30,7 +33,7 @@ private SymbolDeclaredCompilationEvent(SymbolDeclaredCompilationEvent original,
         public ISymbol Symbol { get; }
 
         // PERF: We avoid allocations in re-computing syntax references for declared symbol during event processing by caching them directly on this member.
-        public ImmutableArray<SyntaxReference> DeclaringSyntaxReferences { get; }
+        public ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => _lazyCachedDeclaringReferences.Value;
```

Comment:
This is accessed only during semantic analysis - so if we create compilations in IDE that are used to just query syntax diagnostics, we should theoretically save on these allocations.